sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,332 @@
+            k_idx = tuple(range(self._n_features))
+            k = len(k_idx)
+            k_score = self._calc_score(X, y, k_idx, cloned_estimator)
+            self.subsets_[k] = {
+                'feature_subset_idx': k_idx,
+                'cv_scores': k_score,
+                'avg_score': np.mean(k_score)
+                }
+
+        best_subset = None
+        k_score = 0
+
+        while k != k_to_select:
+            prev_subset = set(k_idx)
+            if self.forward:
+                k_idx, k_score, cv_scores = self._inclusion(
+                    orig_set=orig_set,
+                    subset=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+            else:
+                k_idx, k_score, cv_scores = self._exclusion(
+                    feature_set=prev_subset,
+                    X=X,
+                    y=y,
+                    estimator=cloned_estimator
+                )
+
+            k = len(k_idx)
+            if k not in self.subsets_ or (self.subsets_[k]['avg_score'] <
+                                          k_score):
+                self.subsets_[k] = {
+                    'feature_subset_idx': k_idx,
+                    'cv_scores': cv_scores,
+                    'avg_score': k_score
+                }
+
+        if select_in_range:
+            max_score = float('-inf')
+            for k in self.subsets_:
+                if (k < self.n_features_to_select[0] or
+                        k > self.n_features_to_select[1]):
+                    continue
+                if self.subsets_[k]['avg_score'] > max_score:
+                    max_score = self.subsets_[k]['avg_score']
+                    best_subset = k
+            k_score = max_score
+            k_idx = self.subsets_[best_subset]['feature_subset_idx']
+
+        self.support_ = k_idx
+        self.support_ = self._get_support_mask()
+        self.score_ = k_score
+        return self
+
+    def _calc_score(self, X, y, indices, estimator):
+        if self.cv:
+            scores = cross_val_score(estimator,
+                                     X[:, indices], y,
+                                     cv=self.cv,
+                                     scoring=self._scorer,
+                                     n_jobs=self.n_jobs,
+                                     pre_dispatch=self.pre_dispatch)
+        else:
+            estimator.fit(X[:, indices], y)
+            scores = np.array([self._scorer(estimator,
+                               X[:, indices], y)])
+        return scores
+
+    def _inclusion(self, orig_set, subset, X, y, estimator):
+        all_avg_scores = []
+        all_cv_scores = []
+        all_subsets = []
+        res = (None, None, None)
+        remaining = orig_set - subset
+        if remaining:
+            for feature in remaining:
+                new_subset = tuple(subset | {feature})
```

Comment:
Unless you correct for it elsewhere, if we are reporting the list of indices in an Attribute, I think it will have nondeterministic order due to this. Please test.

Reply:
we could change it to `new_subset = tuple(sorted(subset | {feature}))`, that should fix the issue, right?